### PR TITLE
repart: fixes when operating on unaligned partitions

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #20622

### CODE REVIEW

File: test/units/testsuite-58.sh [hunk 1]
```diff
@@ -84,6 +84,38 @@ cmp /var/tmp/testsuite-58.img /var/tmp/testsuite-58.2.img
...
+
+rm -f /var/tmp/testsuite-58.3.img /tmp/testsuite-58-3.dump
+mkdir -p /tmp/testsuite-58.3-defs/
+
+cat >/tmp/testsuite-58.3-defs/root.conf <<EOF
+[Partition]
+Type=root
+EOF
+
+truncate -s 10g /var/tmp/testsuite-58.3.img
```
Comment (bluca) [inline test/units/testsuite-58.sh:97]: 10g might be a bit harsh on some of the overcrowded testbeds we got - can the same functionality be tested with something smaller?

**Review by bluca — APPROVED** [review]
lgtm, one question about the test

File: test/units/testsuite-58.sh (hunk 1 above)
Comment (poettering) [reply test/units/testsuite-58.sh:97]: it's a sparse file, it's free. (all linux fs know sparse files. i mean, the only common fs we deal with in linux that doesn't do sparse is FAT, but /var/tmp/ is not going to be that)

My goal here was to reuse the precise partition definition from the original bug report, which had such a large disk...

Consider it an extension to the test case: that the assumption that disk images that are sparse are really free holds, and that they are not made unsparse by systemd-repart. ;-)

Comment (poettering) [reply test/units/testsuite-58.sh:97]: (by sparse file i mean: it's entirely empty except the blocks needed for the part table. i.e. a few KB at the beginning of the file and at the end, and everything in between is just one gigantic hole)

Comment (bluca) [reply test/units/testsuite-58.sh:97]: ok, sounds good

**Review by yuwata — APPROVED** [review]

**Pull request merged**